### PR TITLE
[CI Fest] Better processing of broken parts and their projections (fixes rare cases of lost forever)

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
In rare cases ClickHouse could consider parts as broken because of some unexpected projections on disk. Now it's fixed.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit fd2d56d58a6ebfd7c995bfc88fafde75dd12381f with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66898/fd2d56d58a6ebfd7c995bfc88fafde75dd12381f/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91693daf599bbaeaed5adb8cba257c176628e2be/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Storages/MergeTree/DataPartsExchange.cpp [hunk 1]
```diff
@@ -224,14 +225,19 @@ void Service::processQuery(const HTMLForm & params, ReadBuffer & /*body*/, Write
     }
     catch (const Exception & e)
     {
-        if (e.code() != ErrorCodes::ABORTED && e.code() != ErrorCodes::CANNOT_WRITE_TO_OSTREAM)
+        if (e.code() != ErrorCodes::ABORTED
+            && e.code() != ErrorCodes::CANNOT_WRITE_TO_OSTREAM
+            && !isRetryableException(std::current_exception()))
```
Comment (tavplubix) [inline src/Storages/MergeTree/DataPartsExchange.cpp:230]: Should we add these error codes to `isRetryableException`?

**Review by tavplubix — APPROVED** [review]

**Comment by alesapin** [comment]
Timeouts look weirds :(

**Comment by alesapin** [comment]
`00159_parallel_formatting_http` https://github.com/ClickHouse/ClickHouse/issues/67010

**Comment by alesapin** [comment]
`03036_dynamic_read_subcolumns` https://github.com/ClickHouse/ClickHouse/issues/67011

**Comment by alesapin** [comment]
This PR already addressing flaky test, so it's valid to merge with triaged unrelated test failures. 
Bugfix validation will not be fixed -- to hard to add a test for it.

**Pull request merged**